### PR TITLE
support registry mirrors

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/11736

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by kopiczko** [comment]
Tested on gauss. Works both with `quay.io` and `docker.io`. I checked images, all pods and systemd units are up.

**Comment by calvix** [comment]
can we test it also for china?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add `--service.registry.mirrors` flag for setting registry mirror domains.
+- Replace `--service.registrydomain` with `--service.registry.domain`.
```
Comment (xh3b4sd) [inline CHANGELOG.md:14]: Doesn't this mean something changed? It is in the added section.

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +27,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Use `0.1.1` tag for `k8s-api-heahtz` image.
 - Use `0.1.0` tag for `k8s-setup-network-env` image.
 - Use `0.1.0` tag for `aws-attach-etcd-dep` image.
+- Use `0.1.0` tag for `k8s-setup-network-env` image.
+- Use `0.1.1` tag for `k8s-api-heahtz` image.
+
+### Fixed
+
+- Fix failing go template rendering of KMS encryption content.
```
Comment (xh3b4sd) [inline CHANGELOG.md:35]: This here does not look right. It is already mentioned above.

File: CHANGELOG.md (hunk 1 above)
Comment (kopiczko) [reply CHANGELOG.md:14]: I agree but the way we do majors with K8s minors gives me a headache here. Should it be `### Changed` even though we'll release next minor?

File: CHANGELOG.md (hunk 2 above)
Comment (kopiczko) [reply CHANGELOG.md:32]: Thanks, fixed.

**Comment by kopiczko** [comment]
@calvix tested on giraffe.

File: CHANGELOG.md (hunk 1 above)
Comment (kopiczko) [reply CHANGELOG.md:15]: We can fix it after merge. Or even release.

Comment (xh3b4sd) [reply CHANGELOG.md:15]: I know where you are coming from and this screws your automation ideas to some extend. Having changes in the changelog "added" section does just not make any sense semantically.

Comment (xh3b4sd) [reply CHANGELOG.md:15]: So yes, please change that.

Comment (kopiczko) [reply CHANGELOG.md:15]: https://github.com/giantswarm/aws-operator/pull/2585

**Pull request merged**